src/components/createPostModal/index.jsx

```diff
@@ -1,59 +1,69 @@
-import { useState } from "react";
-import useModal from "../../hooks/useModal";
-import "./style.css";
-import Button from "../button";
+import { useState } from 'react'
+import useModal from '../../hooks/useModal'
+import './style.css'
+import Button from '../button'
+import useUser from '../../hooks/useUser'
 
 const CreatePostModal = () => {
-  // Use the useModal hook to get the closeModal function so we can close the modal on user interaction
-  const { closeModal } = useModal();
-
-  const [message, setMessage] = useState(null);
-  const [text, setText] = useState("");
-
-  const onChange = (e) => {
-    setText(e.target.value);
-  };
-
-  const onSubmit = () => {
-    setMessage("Submit button was clicked! Closing modal in 2 seconds...");
-
-    setTimeout(() => {
-      setMessage(null);
-      closeModal();
-    }, 2000);
-  };
-
-  return (
-    <>
-      <section className="create-post-user-details">
-        <div className="profile-icon">
-          <p>AJ</p>
-        </div>
-        <div className="post-user-name">
-          <p>Alex J</p>
-        </div>
-      </section>
-
-      <section>
-        <textarea
-          onChange={onChange}
-          value={text}
-          placeholder="What's on your mind?"
-        ></textarea>
-      </section>
-
-      <section className="create-post-actions">
-        <Button
-          onClick={onSubmit}
-          text="Post"
-          classes={`${text.length ? "blue" : "offwhite"} width-full`}
-          disabled={!text.length}
-        />
-      </section>
-
-      {message && <p>{message}</p>}
-    </>
-  );
-};
-
-export default CreatePostModal;
+    // Use the useModal hook to get the closeModal function so we can close the modal on user interaction
+    const { closeModal } = useModal()
+    const { currentUser } = useUser()
+
+    const [message, setMessage] = useState(null)
+    const [text, setText] = useState('')
+
+    const onChange = (e) => {
+        setText(e.target.value)
+    }
+
+    const onSubmit = () => {
+        setMessage('Submit button was clicked! Closing modal in 2 seconds...')
+
+        setTimeout(() => {
+            setMessage(null)
+            closeModal()
+        }, 2000)
+    }
+
+    const userInitials =
```

Comment:
Can we refactor this to not have so many `ors`? 

We only won't have a `currentUser` firstname/last name if the user isn't logged in - if the user isn't logged in, they shouldn't be able to make a post anyway. So instead, do a guard check on `firstName` and `lastName`, if they don't exist, error / navigate to login page